### ISSUE TITLE
Experiment module: Adding sequential years

### ISSUE THREAD

**Issue by aislinnpearson** [Rothamsted-Ecoinformatics/farm_rothamsted#277]
Most of our experiments are annual but all of the Classicals and LTEs cover multiple years in the same location. For this reason, rather than re-upload the experiment every year (which is what we currently do), it would be nice to submit it for continuation. 

Almost all of these multi-year experiments change: the treatments rotate or in some cases are changed completely. The plot layouts can change, for example when plots are subdivided or merged. Other times the people associated with the experiment can change. 

the way I think this is implemented in Grassroots is that the Programme, Location and Study stay the same, but the trial is updated. Using Broadbalk as an example: the programme is the NBRI/ grant supporting the experiments. The location is the actual location of the experiment (the experiment boundary, field and farm but not the specific plot attributes) and the study is Broadbalk itself. The trial is Broadbalk in a particular year. Our experiment codes work on a similar basis: R/BK/1 is used to refer to Broadbalk the Study, while 2022/R/BK/1 is used to refer to a specific year within that Study. 

My suggestion is that we design the second version of the experiment module along these lines (Programme, Study, Location, Trial). However, as we are not just designing the experiment module for plot trials (see #271) we rename these things _Programme_, _Study_, _Location_ and _Year_. Important to also note that a Study might repeat in a different location the following year (I think WGIN does this). I will create a separate issue for this.

I have labelled this issue as 'Should do' as we could re-upload experiments year on year, and associate them with the same location as a temporary solution.

See also this issue:
- https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/issues/517

**Comment by aislinnpearson** [comment]
I discussed this yesterday with @mstenta and @paul121. To summarise that conversation

There are multiple aspects to adding sequential years (plans) to an experiment. This includes:

1. either duplicating or linking to plot assets
2. either duplicating or linking to the experiment boundary
3. duplicating _and_ linking to an existing plan to show sequential progression over multiple years (the 'study number' on the existing plans)
4. We probably also need to think about sensor assets and other land/plant assets associated with the experiment (e.g. experiment surrounds, paths, etc) which may also remain consistent over time.

There are three use cases we would need to cover as a minimum:

- An experiment which remains in the same place year on year, where the design is the same and the treatments either rotate or remain the same between years (e.g. Broadbalk, Park Grass, LSRE, etc)
- An experiment where the design is the same but it moves every year, and the treatments either remain the same (e.g. annual trials that needs to be repeated at different locations in three sequential years to get statistical power) 
- An experiment where the design is the same but it moves every year, and the treatments remain the same but the factor levels vary (e.g. WGIN or the Watkins Collection where the design is the same but the varieties change every year)

The most extensive part of this discussion centred around the plot assets - specifically if we should duplicate them for every study year (plan) or reference the same assets year on year.

**Duplicating Plots:**
In this instance the plot_id and plot_number would remain the same but the FarmOS plot asset number would change year on year. Duplicating plots will mean the number of plot assets in FarmOS will increase significantly (circa 3,000 - 4,000 new plots per year based on this year's number of plot assets) and there will be no code assigned by FarmOS to show that the plots relate to one another.  Instead, should we decide to go with referencing the same plots in multiple plans, the plot assets can be consolidated _as long as_ the plot_id and experiment code remain _exactly_ the same (used to generate a unique [experiment code]: [plot_number] code). @paul121 made the point that we could use the ISGNs either as a tag or as a dedicated field once we are able to mint these.
This is by far the easiest option so we decided to stick with it for the MVP so we agreed that we would make it possible to duplicate plots rather than reference plots following the specification in #358.

**Referencing plots**
This would require removing the association between the plot assets and the variables on the plot asset (this association would instead only be maintained on the plans). The problem with this as @paul121 pointed out is that if you edit the geometry in one year, it would affect the plots in all years. For this reason, we might want to make it very difficult to edit a plot's geometry after records have been assigned to it. This would mean fewer plot assets in FarmOS, but more work for @paul121 in terms of the coding the work.
As an aside, @mstenta wasn't concerned about the volume of plot assets in terms of performance - the only thing it might affect is visualisations, especially as the plots are stacked on top of each other year after year on the map.

**A third option?**
A third option which we mentioned but didn't discuss in detail is the idea of creating a new entity that links the plots and the plans. This could also solve the problem of linking between annual crop plans and sensors/ experiment boundaries and other assets, and in fact @mstenta mentioned it is similar to an issue relating to plans and assets in version 1.x, although tehre it wasn't just a reference but rather a reference with additional metadata.  This is similar to issue:
- #103 

**In Summary**
For 2.10.x (the MVP) there won't be an option to duplicate plans or assets (it will all have to be done manually). In the short term we could probably build a 'duplicate' feature, but this would need to be easier to do in batches (so the user chooses to duplicate the plan, the plots and any other assets separately) for which I have created a seperate issue (#377). However, resolving the issue fully needs more funding and so is best dealt with in the subsequent funding cycle.

**Comment by aislinnpearson** [comment]
This was resolved by having the new plans for every year of the experiment
